docs/user-guide/dev-environment.md

```diff
@@ -420,6 +420,135 @@ access to software libraries if available.
     be found in the [Software libraries](../software-libraries/)
     section of the user guide.
 
+## Switching to a different HPE Cray Programming Environment release
+
+!!! important
+    See the section below on using non-default versions of HPE Cray libraries
+    below as this process will generally need to be followed when using software
+    from non-default PE installs.
+
+Access to non-default PE environments is controlled by the use of the `cpe` modules.
+These modules are typically loaded *after* you have restored a PrgEnv and loaded all
+the other modules you need and will 
+set your compile environment to match that in the other PE release. This means:
+
+- The compiler version will be switched to the one from the selected PE
+- Libraries (or modules) that are loaded will have their verison
+  switched to the ones from the selected PE.
+
+Some things are not changed on loading the `cpe` module. In particular:
+
+- The default version of modules accessed by the `module` command are not
+  updated to those in the specified PE. You should ensure all modules are
+  loaded *before* you load the `cpe` module to ensure all modules get switched
+  to new versions.
+
+As noted above, you will also need to setup your environment to use non-default
+versions of HPE Cray libraries (including those changed by the `cpe` module) as
+described in the section below.
+
+## Using non-default versions of HPE Cray libraries on ARCHER2
+
+If you wish to make use of non-default versions of libraries provided by HPE
+Cray (usually because they are part of a non-default PE release: either old
+or new) then you need to make changes at *both* compile and runtime. In summary,
+you need to load the correct module and also make changes to the `LD_LIBRARY_PATH`
+environment variable.
+
+**At compile time** you need to load the version of the library module before you compile
+*and* set the LD_LIBRARY_PATH environment variable to include the contencts of
+`$CRAY_LD_LIBRARY_PATH` as the first entry. For example, to use the, non-default, 20.08.1.2
+version of HPE Cray LibSci in the default programming environment (Cray Compiler Environment,
+CCE) you would first setup the environment to compile with:
+
+```
+auser@uan01:~/test/libsci> module swap cray-libsci cray-libsci/20.08.1.2 
+auser@uan01:~/test/libsci> export LD_LIBRARY_PATH=$CRAY_LD_LIBRARY_PATH:$LD_LIBRARY_PATH
+```
+
+The order is important here: every time you change a module, you will need to reset
+the value of `LD_LIBRARY_PATH` for the process to work (it will not be updated
+automatically).
+
+Now you can compile your code. You can check that the executable is using the correct version 
+of LibSci with the `ldd` command and look for the line beginning `libsci_cray.so.5`, you
```

Comment:
This is specific to cray-libsci only. As we're now moving to loading new CPEs, it might be worth making this particular part more general (perhaps by replacing the `libsci_cray.so.5` part with the tip about there being no `*/lib64/*`)?

Reply:
I am not sure I get what you mean here @jsindt. This is true of all non-default Cray-provided libraries I think. LibSci here is used as an example to illustrate the issue not to say it just applies to libsci. Can you perhaps suggest some specific text that would improve the description?

Reply:
I think that what you've done to address @kevinstratford's points have addressed my points as well.

Reply:
Ta, I will merge this now then.